javascript/modules/client/src/client-erc20.ts

```diff
@@ -434,21 +432,140 @@ export class ClientErc20 extends ClientCore implements IClientErc20 {
+        startDate,
+        endDate,
+        erc20VotingProposal.executed,
+        erc20VotingProposal
+          .yea,
+        erc20VotingProposal.nay,
+      );
+      // get start and end date
+      return {
+        id: erc20VotingProposal.id,
+        dao: {
+          address: erc20VotingProposal.dao.id,
+          name: erc20VotingProposal.dao.name,
+        },
+        creatorAddress: erc20VotingProposal.creator,
+        metadata: {
+          title: metadata.title,
+          summary: metadata.summary,
+          description: metadata.description,
+          resources: metadata.resources,
+          media: metadata.media,
+        },
+        startDate,
+        endDate,
+        creationDate,
+        actions: erc20VotingProposal.actions.map(
+          (action: { data: string; to: string; value: string }): DaoAction => {
+            return {
+              data: hexToBytes(strip0x(action.data)),
+              to: action.to,
+              value: BigInt(action.value),
+            };
+          },
+        ),
+        status,
+        result: {
+          yes: BigInt(erc20VotingProposal.yea),
+          no: BigInt(erc20VotingProposal.nay),
+          abstain: BigInt(erc20VotingProposal.abstain),
+        },
+        settings: {
+          // TODO
+          // this should be decoded using the number of decimals that we want
+          // right now the encoders/recoders use 2 digit precission but the actual
+          // subgraph values are 18 digits precision. Uncomment below for 2 digits
+          // precision
+
+          // minSupport: decodeRatio(
+          //   BigInt(erc20VotingProposal.supportRequiredPct),
+          //   2,
+          // ),
+          // minTurnout: decodeRatio(
+          //   BigInt(erc20VotingProposal.participationRequiredPct),
+          //   2,
+          // ),
+          // TODO DELETE ME
+          minSupport: parseFloat(
+            formatEther(erc20VotingProposal.supportRequiredPct),
+          ),
+          minTurnout: parseFloat(
+            formatEther(erc20VotingProposal.participationRequiredPct),
+          ),
+          duration: parseInt(erc20VotingProposal.endDate) -
+            parseInt(erc20VotingProposal.startDate),
+        },
+        token: {
+          address: erc20VotingProposal.pkg.token.id,
+          symbol: erc20VotingProposal.pkg.token.symbol,
+          name: erc20VotingProposal.pkg.token.name,
+          decimals: parseInt(erc20VotingProposal.pkg.token.decimals),
+        },
+        usedVotingWeight,
+        totalVotingWeight: BigInt(erc20VotingProposal.votingPower),
+        votes: erc20VotingProposal.voters.map(
+          (voter: { voter: { id: string }; vote: string; weight: string }) => {
+            let vote;
+            switch (voter.vote) {
+              case "Yea":
+                vote = VoteValues.YES;
+                break;
+              case "Nay":
+                vote = VoteValues.NO;
+                break;
+              case "Abstain":
```

Comment:
I think the string values changed? Otherwise, let's factor this in an enum so that we only need to change things once, whenever subgraph is updated to use "yes" "no", etc